### PR TITLE
Initial a new module

### PR DESCRIPTION
**Pull request by Wanghx0991**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+variable "payment_type" {
+  type = string
+  description = "The payment type of the resource. Valid values: Subscription."
+  default = "Subscription"
+}
+
+variable "period" {
+  type = number
+  description = "The period. Unit: months. Valid values: 1,3, 6, 12, 24, 36."
+  default = null
```
Comment (xiaozhu36) [inline variables.tf]: default = 1

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+variable "plan_id" {
+  type = string
+  description = "The ID of the plan. You can use the alicloud_simple_application_server_plans to query all the plans provided by Simple Application Server in the specified region."
+  default = null
+}
+
+variable "instance_status" {
+  type = string
+  description = "The status of the simple application server. Valid values: Resetting, Running, Stopped."
+  default = null
```
Comment (xiaozhu36) [inline variables.tf]: default = Running

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+variable "description" {
+  type = string
+  description = "The description of the Custom Image."
+  default = null
+}
+
+variable "image_status" {
+  type = string
+  description = "The Shared status of the Custom Image. Valid values: Share, UnShare."
+  default = null
```
Comment (xiaozhu36) [inline variables.tf]: default = "UnShare"

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+variable "auto_renew" {
+  type = bool
+  description = "Specifies whether to enable auto-renewal. Unit: months. Valid values: true and false."
+  default = false
+}
+variable "auto_renew_period" {
+  type = number
+  description = "The auto renew period. Valid values: 1,3, 6, 12, 24, 36. NOTE: The attribute auto_renew is valid when the attribute is true."
+  default = null
+}
```
Comment (xiaozhu36) [inline variables.tf:43]: 没有执行 terraform fmt

File: outputs.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
+output "instance_id" {
+  description = "The Instance ID"
+  value       = concat(alicloud_simple_application_server_instance.instance.*.id, [""])[0]
+}
+
+output "instance_name" {
+  description = "The Instance ID"
```
Comment (xiaozhu36) [inline outputs.tf]: The instance Name

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,120 @@
+variable "create_instance" {
+  type = bool
+  description = "Shether to create the alicloud_simple_application_server_instance"
```
Comment (xiaozhu36) [inline variables.tf]: Whether to create a new simple application server instance.

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  default = false
+}
+
+variable "create_snapshot" {
+  type = bool
+  description = "Shether to create the alicloud_simple_application_server_snapshot"
```
Comment (xiaozhu36) [inline variables.tf]: Whether to create a new simple application server snapshot.

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  default = false
+}
+
+variable "create_image" {
+  type = bool
+  description = "Shether to create the alicloud_simple_application_server_custom_image"
```
Comment (xiaozhu36) [inline variables.tf]: Whether to create a new simple application server custom image.

File: variables.tf [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  default = false
+}
+
+variable "instance_id" {
+  type = string
+  description = "the alicloud_simple_application_server_instance ID"
```
Comment (xiaozhu36) [inline variables.tf]: An existing simple application server instance id. It will be ignored when `create_instance = true`

File: variables.tf [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  default = ""
+}
+
+variable "system_snapshot_id" {
+  type = string
+  description = "the System Snapshot ID"
```
Comment (xiaozhu36) [inline variables.tf]: An existing simple application server instance id. It will be ignored when `create_snapshot = true`

File: variables.tf [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+  default = null
+}
+variable "data_disk_size" {
+  type = number
+  description = "The size of the data disk. Unit: GB. Valid values: 0 to 16380."
+  default = null
+}
+variable "image_id" {
+  type = string
+  description = " The ID of the image. You can use the alicloud_simple_application_server_images to query the available images in the specified region. The value must be an integral multiple of 20."
```
Comment (xiaozhu36) [inline variables.tf]: An existing simple application server image id. It will be ignored when create_image = true

File: variables.tf [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+variable "snapshot_name" {
+  type = string
+  description = " The name of the snapshot. The name must be 2 to 50 characters in length. It must start with a letter and cannot start with http:// or https://. It can contain letters, digits, colons (:), underscores (_), periods (.),and hyphens (-)."
+  default = null
+}
+
+#alicloud_simple_application_server_custom_image
+variable "custom_image_name" {
+  type = string
+  description = "The name of the resource. The name must be 2 to 128 characters in length. It must start with a letter or a number. It can contain letters, digits, colons (:), underscores (_) and hyphens (-)."
```
Comment (xiaozhu36) [inline variables.tf]: The name of a new custom image. xxxx

File: main.tf [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+  instance_id        = var.instance_id != "" ? var.instance_id : concat(alicloud_simple_application_server_instance.instance.*.id, [""])[0]
+  system_snapshot_id = var.system_snapshot_id != "" ? var.system_snapshot_id : concat(alicloud_simple_application_server_snapshot.snapshot.*.id, [""])[0]
+}
+
+resource "alicloud_simple_application_server_instance" "instance" {
+  count             = var.create_instance ? 1 : 0
+  auto_renew        = var.auto_renew
+  auto_renew_period = var.auto_renew_period
+  data_disk_size    = var.data_disk_size
+  image_id          = var.image_id
```
Comment (xiaozhu36) [inline main.tf:11]: 如果创建了一个新的custom image，这个地方应该用创建出来的image 的id。

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: main.tf (hunk 13 above)
Comment (Wanghx0991) [reply main.tf:11]: 用两个参数区分了, image_id和custom_image_id, 创建的时候可以分别指定

File: outputs.tf (hunk 5 above)
Comment (Wanghx0991) [reply outputs.tf]: Done

**Pull request merged**